### ISSUE TITLE
Please, don't flush the disk continuously

### ISSUE THREAD

**Issue by Anyeos** [johannesjo/super-productivity#172]
I noted that superProductivity flush the disk each second when I have a running task.
That is just contradictory because it is not very productive. Because when I do some disk demanding task like processing a video or building a software it just drain a little of the speed in the process (because random disk access).
At the same time it reduces the lifetime of my hard disk when I am doing nothing with the disk but only running a task in superProductivity.

I suggest that put a more long time for flushing to disk. Maybe every minute or selectable as a setting.

I tried the newer version of superProductivity (superProductivity-2.6.3-x86_64.AppImage) and the problem still exists. So it is something that not been changed.

**Comment by johannesjo** [comment]
Not really sure what could cause this. Maybe it's an electron thing.

Do you really mean flushing, like in:

> All data is immediately written to a persistent log on the filesystem without necessarily flushing to disk. In effect this just means that it is transferred into the kernel's pagecache.

**Comment by johannesjo** [comment]
After digging into this I find that there is nothing unusual going on. Date must be saved while tracking.  Sure you could do everything in  memory and only save on starting or closing the app, but that would mean that data can be lost and it would complicate development by quite a bit. If you have an idea to improve dealing with this issue, please let me know.

**Comment by cemsbr** [comment]
I noticed the every-second disk usage stops when I stop the pomodoro timer. As soon as I resume the timer, it writes to disk every second. What is it doing? Writing the new remaining time every second?

**Comment by johannesjo** [comment]
@cemsbr backups are made every two seconds when data is changed (which it does, when tracking time). I am not sure about the inner workings of indexeddb but I would assume that it also writes to the disk if backups would be disabled, which would happen every second.

**Comment by cemsbr** [comment]
Is there a way to not modify the data every second, i.e. making the timer just a calculation that uses only the start time as data (and current time)?

**Comment by johannesjo** [comment]
@cemsbr this is theoretically possible, but would – as far as I can see – introduce a lot of complexity and likely some expected (e.g.special cases for syncing) and some unexpected problems. Data needs to be written to the disk at some point and if it isn't, there might be data loss in case of a crash or something. We'd better have a *very* good reason for this. :)

**Comment by Ekleog** [comment]
Just to say that I just noticed something similar after my monitoring started firing alerts: when tracking time to any task, super-productivity starts writing megabytes per second to the disk. This is probably something that'll end up killing hard disks, as this amount of data writes is pretty significant.

I don't think anyone would feel sad about losing one or two minutes of time tracking, so maybe a reasonable solution would be to flush to disk on each manual change, but only every few minutes when the only changes are time tracking?

(Also, I see you added labels and changed the title on this task after it was closed, maybe it could be reopened?)

**Comment by johannesjo** [comment]
@Ekleog Conceptually this makes sense to me. I am a bit worried about all the technical implications this would have (we would need to add some intermediary in memory layer for this to work or to completely change how storing data works).

**Comment by Anyeos** [comment]
Hello. I just want to comment something but it doesnt mean that I want you to do or change something. It is only my way of thinking.

I not used anymore this app because the above. And sorry but that is why I does not come here anymore. 
Instead I developed a very basic app for my personal use and it saves to the disk only when I start the timer. My App saves the start timestamp and if the App crash or the computer turns off or something goes wrong the timestamp just mark the exact time of when the timer started to count. When I reopen the app, all the other data are automatically computed from the timestamp. And all the values are updated no needing to write to the disk again. 
That is the behaviour that I adopted to my basic tasks desktop application. 

And I dont understand how Super Productivity works with that, but I dont understand neither why it is like that. Sorry but it make me think that this app is bad planned from the begining.

And I repeat, it is only my opinion I may be wrong. I dont know why it is done like that.
